lib/src/rules/avoid_types_as_parameter_names.dart

```diff
@@ -66,20 +67,47 @@ class _Visitor extends SimpleAstVisitor<void> {
       var identifier = parameter.identifier;
       if (declaredElement != null &&
           declaredElement is! FieldFormalParameterElement &&
-          declaredElement.hasImplicitType &&
           identifier != null &&
           _isTypeName(node, identifier)) {
         rule.reportLint(identifier);
       }
     }
   }
 
+  @override
+  void visitGenericFunctionType(GenericFunctionType node) {
```

Comment:
We don't need this either. An instance of `GenericFunctionType` contains an instance of `FormalParameterList`, and that instance will be visited by `visitFormalParameterList`. This also causes the parameters to be visited twice.

Reply:
Thanks for your patience.

Clear. I probably misunderstood your other comment.

There was a check at the very beginning of `visitFormalParameterList` to check the parent, now removed in the last commit.

I hope it is now fine.